### ISSUE TITLE
Meta Issue for k8s 1.28 bump

### ISSUE THREAD

**Issue by acornett21** [operator-framework/operator-sdk#6554]
## Meta Issue for k8s 1.28 bump
In order to bump Operator SDK to support Kubernetes 1.28 there are a few dependencies we rely on making the bump first.
This issue is meant to help track all dependencies and the status of their bumps.

### Order of dependency bumps
**Step 1: upgrade controller-runtime, controller-tools, helm, and operator-framework/operator-manifest-tools(can be done in parallel)**
* [x]  sigs.k8s.io/controller-runtime
* [x]  sigs.k8s.io/controller-tools
* [x]  operator-framework/operator-manifest-tools

**Step 2: upgrade operator-framework/api and sigs.k8s.io/kubebuilder-declarative-pattern (can be done immediately after controller-runtime)**
* [x]  operator-framework/api
* [x]  sigs.k8s.io/kubebuilder-declarative-pattern

**Step 3.1: upgrade operator-framework/operator-registry and operator-framework/operator-lib (can be done immediately after operator-framework/api)**
* [x]  operator-framework/operator-registry
* [x]  operator-framework/operator-lib

**Step 3.2: upgrade kubebuilder (blocked until kubebuilder-declarative-pattern is bumped)**
* [x]  sigs.k8s.io/kubebuilder/v3

**Step 4: upgrade operator-framework/helm-operator-plugins, operator-framework/java-operator-plugins, and operator-framework/ansible-operator-plugins**
* [x]  operator-framework/helm-operator-plugins
* [x]  operator-framework/java-operator-plugins
* [x]  operator-framework/ansible-operator-plugins

**Step 5: upgrade operator-framework/operator-sdk dependencies**
* [x]  Bump Ginkgo/v2 and Kubebuilder 1.28
* [x]  Bump SDK to use k8s 1.28
* [x]  ~~Bump envtest to 1.28~~
    *  this is no longer needed as we automatically get the latest version: https://github.com/operator-framework/operator-sdk/blob/master/Makefile#L170

**Comment by acornett21** [comment]
- Relates: kubernetes-sigs/kubebuilder-declarative-pattern#349

**Comment by Neo2308** [comment]
Current releases that have k8s updates:

Controller-runtime: https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.16.0 (latest is v0.16.3)
Controller-tools: https://github.com/kubernetes-sigs/controller-tools/releases/tag/v0.13.0
Kubebuilder: https://github.com/kubernetes-sigs/kubebuilder/releases/tag/v3.12.0

**Comment by Neo2308** [comment]
Relates to https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/352

**Comment by acornett21** [comment]
kubebuilder `v3.13.0` should address this.

https://github.com/kubernetes-sigs/kubebuilder/releases/tag/v3.13.0

**Comment by acornett21** [comment]
`java-operator-plugin` `v0.9.0` has the needed changes.

https://github.com/operator-framework/java-operator-plugins/releases/tag/v0.9.0

**Comment by jberkhahn** [comment]
Need to keep this in mind: #6635

**Comment by acornett21** [comment]
`ansible-operator-plugin` `v1.34.0` has the needed changes.
https://github.com/operator-framework/ansible-operator-plugins/releases/tag/v1.34.0

**Comment by Neo2308** [comment]
@acornett21/ @oceanc80 When is the 1.34 release planned?

**Comment by acornett21** [comment]
Hi @Neo2308 sorry for the delay, I'm not a maintainer, I just contribute, so I'm not sure when a release is planned. 

@everettraven @oceanc80 Do you know when the next release is planned? Thanks!
